core/src/Kokkos_View.hpp

```diff
@@ -2396,6 +2396,120 @@ KOKKOS_INLINE_FUNCTION DeducedCommonPropsType<Views...> common_view_alloc_prop(
   return DeducedCommonPropsType<Views...>(views...);
 }
 
+// -----------------------------------------------------------
+// public is_assignable
+// expose underlying view mapping properties case where layout isn't stride
+template <class DstViewType, class SrcViewType, class Enabled = void>
+struct is_always_assignable {
+  enum : bool { value = false };
+};
+
+template <class DstViewType, class SrcViewType>
+struct is_always_assignable<
+    DstViewType, SrcViewType,
+    typename std::enable_if<
+        (!(std::is_same<typename SrcViewType::traits::array_layout,
+                        Kokkos::LayoutStride>::value) &&
+         // Added to have a new specialization for SrcType of
+         // LayoutStride
+         // default mappings
+         std::is_same<typename DstViewType::traits::specialize, void>::value &&
+         std::is_same<typename SrcViewType::traits::specialize, void>::value &&
+         (
+             // same layout
+             std::is_same<typename DstViewType::traits::array_layout,
+                          typename SrcViewType::traits::array_layout>::value ||
+             // known layout
+             ((std::is_same<typename DstViewType::traits::array_layout,
+                            Kokkos::LayoutLeft>::value ||
+               std::is_same<typename DstViewType::traits::array_layout,
+                            Kokkos::LayoutRight>::value ||
+               std::is_same<typename DstViewType::traits::array_layout,
+                            Kokkos::LayoutStride>::value) &&
```

Comment:
Move this logic into an `Impl::IsKnownLayout` (or something) type trait class template. That seems like readily reusable information, and it'll make this much more readable.